crypto/src/contribution.rs

```diff
@@ -224,14 +381,10 @@ impl Contribution {
     }
 
     #[instrument(level = "info", skip_all, fields(n1=self.g1_powers.len(), n2=self.g2_powers.len()))]
-    pub fn subgroup_check(&self) {
-        assert!(self.pubkey.is_in_correct_subgroup_assuming_on_curve());
-        self.g1_powers
-            .par_iter()
-            .for_each(|point| assert!(g1_subgroup_check(point)));
-        self.g2_powers
-            .par_iter()
-            .for_each(|point| assert!(g2_subgroup_check(point)));
+    pub fn subgroup_check(&self) -> bool {
+        self.pubkey.is_in_correct_subgroup_assuming_on_curve()
+            && !self.g1_powers.par_iter().any(|p| !g1_subgroup_check(p))
+            && !self.g2_powers.par_iter().any(|p| !g2_subgroup_check(p))
```

Comment:
This should return Result and ideally point at the failure. But let's leave for future PR.